azurerm/internal/services/cosmos/tests/resource_arm_cosmosdb_gremlin_graph_test.go

```diff
@@ -0,0 +1,305 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMCosmosDbGremlinGraph_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cosmosdb_gremlin_graph", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMCosmosDbGremlinGraphDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMCosmosDbGremlinGraph_basic(data),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testCheckAzureRmCosmosDbGremlinGraphExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMCosmosDbGremlinGraph_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_cosmosdb_gremlin_graph", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMCosmosDbGremlinGraphDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMCosmosDbGremlinGraph_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRmCosmosDbGremlinGraphExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMCosmosDbGremlinGraph_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_cosmosdb_gremlin_graph"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMCosmosDbGremlinGraph_customConflictResolutionPolicy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cosmosdb_gremlin_graph", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMCosmosDbGremlinGraphDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMCosmosDbGremlinGraph_customConflictResolutionPolicy(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRmCosmosDbGremlinGraphExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMCosmosDbGremlinGraph_indexPolicy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cosmosdb_gremlin_graph", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMCosmosDbGremlinGraphDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMCosmosDbGremlinGraph_indexPolicy(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRmCosmosDbGremlinGraphExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMCosmosDbGremlinGraph_complete(t *testing.T) {
```

Comment:
This is an update test
```suggestion
func TestAccAzureRMCosmosDbGremlinGraph_update(t *testing.T) {
```